h/static/scripts/permissions.coffee

```diff
@@ -37,10 +37,15 @@ module.exports = ['session', (session) ->
   # Typical use: annotation.permissions = permissions.private()
   ###
   private: ->
-    read: [session.state.userid]
-    update: [session.state.userid]
-    delete: [session.state.userid]
-    admin: [session.state.userid]
+    if not session?.state?.userid
```

Comment:
This is overly defensive. If `session` doesn't exist something is badly wrong. And the `session` service sets `state` to an empty object by default. `if not session.state.userid` should be fine here.